scripts/setup-helper-functions.sh

```diff
@@ -37,6 +37,84 @@ function github_checkout {
   cd "${DIRNAME}"
 }
 
+
+# get_cxx_flags [$CPU_ARCH]
+# Sets and exports the variable VELOX_CXX_FLAGS with appropriate compiler flags.
+# If $CPU_ARCH is set then we use that else we determine best possible set of flags
+# to use based on current cpu architecture.
+# The goal of this function is to consolidate all architecture specific flags to one
+# location.
+# The values that CPU_ARCH can take are as follows:
+#   arm64  : Target Apple silicon.
+#   aarch64: Target general 64 bit arm cpus.
+#   avx:     Target Intel CPUs with AVX.
+#   sse:     Target Intel CPUs with sse.
+# Echo's the appropriate compiler flags which can be captured as so
+# CXX_FLAGS=$(get_cxx_flags) or
+# CXX_FLAGS=$(get_cxx_flags "AVX")
+
+function get_cxx_flags {
+  local CPU_ARCH=$1
+
+  local OS
+  OS=`uname`
+  local MACHINE
+  MACHINE=`uname -m`
+
+  if [ -z "$CPU_ARCH" ]; then
+
+    if [ $OS = "Darwin" ]; then
+
+      if [ $MACHINE = "x86_64" ]; then
+        local CPU_CAPABILITIES
+        CPU_CAPABILITIES=`sysctl -a | grep machdep.cpu.features | awk '{print tolower($0)}'`
+
+        if [[ $CPU_CAPABILITIES =~ "avx" ]]; then
+          CPU_ARCH="avx"
+        else
+          CPU_ARCH="sse"
+        fi
+
+      elif [[ `sysctl -a | grep machdep.cpu.brand_string` =~ "Apple" ]]; then
+        # Apple silicon.
+        CPU_ARCH="arm64"
+      fi
+    else [ $OS = "Linux" ];
+
+      local CPU_CAPABILITIES
+      CPU_CAPABILITIES=`cat /proc/cpuinfo | grep flags | head -n 1| awk '{print tolower($0)}'`
+
+      if [[ $CPU_CAPABILITIES =~ "avx" ]]; then
+            CPU_ARCH="avx"
+      elif [[ $CPU_CAPABILITIES =~ "sse" ]]; then
+            CPU_ARCH="sse"
+      elif [ $MACHINE = "aarch64" ]; then
+            CPU_ARCH="aarch64"
+      fi
+    fi
+  fi
+
+  case $CPU_ARCH in
```

Comment:
if the input `CPU_ARCH` is invalid or incorrectly typed, should we throw a warning message that an invalid option was specified?